src/app/login/login-form.component.ts

```diff
@@ -3,30 +3,61 @@ import { CouchService } from '../shared/couchdb.service';
 import { Router, ActivatedRoute } from '@angular/router';
 import { UserService } from '../shared/user.service';
 import { switchMap } from 'rxjs/operators';
+import { FormControl, FormGroup, FormBuilder, Validators } from '@angular/forms';
+import { CustomValidators } from '../validators/custom-validators';
 
 @Component({
   templateUrl: './login-form.component.html',
   styleUrls: [ './login.scss' ]
 })
 export class LoginFormComponent {
+  public userForm: FormGroup;
   constructor(
     private couchService: CouchService,
     private router: Router,
     private route: ActivatedRoute,
-    private userService: UserService
-  ) { }
+    private userService: UserService,
+    private formBuilder: FormBuilder
+  ) {
+       this.userForm = this.formBuilder.group({
+       name: [ '', Validators.compose([
+         Validators.required
+         ]) ],
+       password: [ '', Validators.compose([
+         Validators.required,
+         CustomValidators.matchPassword('repeatPassword', false)
+         ]) ],
+       repeatPassword: [ '', Validators.compose([
+         Validators.required,
+         CustomValidators.matchPassword('password', true)
+         ]) ]
+       });
+  }
 
   createMode: boolean = this.router.url.split('?')[0] === '/login/newuser';
   returnUrl = this.route.snapshot.queryParams['returnUrl'] || '/';
-  model = { name: '', password: '', repeatPassword: '' };
   message = '';
 
   onSubmit() {
     if (this.createMode) {
-      this.createUser(this.model);
+      if (this.userForm.valid) {
+        this.createUser(this.userForm.value.name, this.userForm.value.password, this.userForm.value.repeatPassword);
+      } else {
+         Object.keys(this.userForm.controls).forEach(field => {
+         const control = this.userForm.get(field);
+         control.markAsTouched({ onlySelf: true });
+         });
+      }
     } else {
-      this.login(this.model, false);
-    }
+        if (this.userForm.value.password !== '' && this.userForm.value.name !== '') {

```

Comment:
If we are using reactive forms, you should use that validation.  Please review the [documentation](https://angular.io/guide/reactive-forms) for a solution.